### PR TITLE
Enhancement/polyline-arrow-symbol

### PR DESCRIPTION
**Pull request by Ahmedhossamdev**
This PR introduces the addArrowToPolyline function, which adds a red arrow to a polyline on Google Maps like android app. This can help users easily see the direction of the bus, improving the overall ux. The arrow repeats every 50 pixels along the polyline, and it can be adjusted.

### CODE REVIEW

**Comment by Ahmedhossamdev** [comment]
without-arrow

To

with-arrow

File: src/components/map/RouteMap.svelte [hunk 1]
```diff
@@ -54,6 +55,25 @@
 		}
 	}
 
+	function addArrowToPolyline() {
```
Comment (aaronbrethorst) [inline src/components/map/RouteMap.svelte:58]: I think this should be moved into `src/lib/googleMaps.js`, and take the polyline as a param.

**Review by aaronbrethorst — CHANGES_REQUESTED** [review]
nice addition. please move the code into the googleMaps.js file as indicated!

File: src/components/map/RouteMap.svelte (hunk 1 above)
Comment (Ahmedhossamdev) [reply src/components/map/RouteMap.svelte]: Done

File: src/lib/googleMaps.js [hunk 2]
```diff
@@ -154,10 +156,33 @@ export async function createPolyline(shape) {
...
 
 	return polyline;
 }
+
+/**
+ * Adds an arrow to the polyline.
+ * @param {google.maps.Polyline} polyline - The polyline to which the arrow will be added.
+ */
+export function addArrowToPolyline(polyline) {
+	const arrowSymbol = {
```
Comment (tarunsinghofficial) [inline src/lib/googleMaps.js:172]: Adding a check at the beginning of the function to validate the `polyline` parameter will be great, right @Ahmedhossamdev @aaronbrethorst? Something like this:

```
if (!(polyline instanceof google.maps.Polyline)) {
        console.error('Invalid polyline');
        return;
}
```

File: src/lib/colors.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export const COLORS = {
```
[the remark this answers is not in the document]
Comment (Ahmedhossamdev) [reply src/lib/colors.js:1]: awesome! I will try it

File: src/lib/googleMaps.js (hunk 2 above)
Comment (Ahmedhossamdev) [reply src/lib/googleMaps.js:172]: Yep, that’s a great idea for checking the polyline parameter. Moving to TypeScript could also improve type safety and help as the project grows.

**Review by aaronbrethorst — APPROVED** [review]
Nice work! 👍

**Pull request merged**